### PR TITLE
spl-token: Enums for Cli command names

### LINKED ISSUE #2598: [spl-token] add enum for cli command names
problem: currently, all cli commands use strings like "create-token" which are hardcoded at different places. difficult to maintain

fix: introduce command name enum that can be turned into string

cc: @CriesofCarrots

### PR DESCRIPTION
**Pull request by atharmohammad**
Fixes #2598 
Added Enums for Command Names and replaced hard coded command names
@joncinque

### CODE REVIEW

File: token/cli/src/main.rs [hunk 1]
```diff
@@ -2535,6 +2568,7 @@ fn main() -> Result<(), Error> {
     let mut multisigner_ids = Vec::new();
 
     let (sub_command, sub_matches) = app_matches.subcommand();
+    let sub_command_ = CommandName::from_str(sub_command).unwrap();
```
Comment (joncinque) [inline token/cli/src/main.rs:2571]: nit: call this `sub_command`

File: token/cli/src/main.rs [hunk 2]
```diff
@@ -100,7 +101,39 @@ pub const MULTISIG_SIGNER_ARG: ArgConstant<'static> = ArgConstant {
...
+    Approve,
+    Revoke,
+    Balance,
+    Supply,
+    Accounts,
+    Address,
+    AccountInfo,
+    MultisigInfo,
+    Gc,
+    SyncNative
```
Comment (joncinque) [inline token/cli/src/main.rs:130]: Eventually, the idea for these is to be completely expressive with the values required for the command encoded as well.

For example, check out the CLI commands in the main repo: https://github.com/solana-labs/solana/blob/c419845cfeb1c7ab9ca959de66e90b9dacbc25b7/cli/src/cli.rs#L47

But we can definitely start here :smile:

File: token/cli/src/main.rs [hunk 3]
```diff
@@ -100,7 +101,39 @@ pub const MULTISIG_SIGNER_ARG: ArgConstant<'static> = ArgConstant {
...
+}
+
+pub fn to_str(command:CommandName) -> &'static str {
+    let command_str : &'static str = command.into();
+    command_str
+}
```
Comment (joncinque) [inline token/cli/src/main.rs:136]: Let's remove this function and just call `.into()` on the `CommandName` directly

File: token/cli/src/main.rs [hunk 4]
```diff
@@ -3050,7 +3084,7 @@ fn process_command(
                 bulk_signers,
             )
         }
-        ("sync-native", arg_matches) => {
+        (CommandName::SyncNative, arg_matches) => {
             let address = config.associated_token_address_for_token_or_override(
                 arg_matches,
                 "address",
```
Comment (joncinque) [inline token/cli/src/main.rs:3090]: I can't comment lower, but can you remove the final case?
```
_ => unreachable!(),
```
With this change, we can be explicit!

**Review by joncinque — COMMENTED** [review]
Great work!  A few small points, then we should be able to get this integrated

File: token/cli/src/main.rs [hunk 5]
```diff
@@ -3264,19 +3291,24 @@ mod tests {
             &multisig_member_help,
         )
         .get_matches_from(args);
-        let (sub_command, sub_matches) = app_matches.subcommand();
+        let (sub_command_, sub_matches) = app_matches.subcommand();
+        let sub_command = CommandName::from_str(sub_command_).unwrap();
```
Comment (joncinque) [inline token/cli/src/main.rs:3295]: nittiest of nits: in Rust, you can rebind an old variable name, so no need to have the underscore.  You can do
```
 let (sub_command, sub_matches) = app_matches.subcommand();
let sub_command = CommandName::from_str(sub_command).unwrap();
```

File: token/cli/src/main.rs [hunk 6]
```diff
@@ -2534,7 +2561,8 @@ fn main() -> Result<(), Error> {
     let mut bulk_signers: Vec<Box<dyn Signer>> = Vec::new();
     let mut multisigner_ids = Vec::new();
 
-    let (sub_command, sub_matches) = app_matches.subcommand();
+    let (sub_command_, sub_matches) = app_matches.subcommand();
+    let sub_command = CommandName::from_str(sub_command_).unwrap();
```
Comment (joncinque) [inline token/cli/src/main.rs:2565]: same here

File: token/cli/src/main.rs [hunk 7]
```diff
@@ -3060,16 +3088,15 @@ fn process_command(
...
 }
 
-fn format_output<T>(command_output: T, command_name: &str, config: &Config) -> String
+fn format_output<T>(command_output: T, command_name: &CommandName, config: &Config) -> String
 where
     T: Serialize + Display + QuietDisplay + VerboseDisplay,
 {
     config.output_format.formatted_string(&CommandOutput {
-        command_name: String::from(command_name),
+        command_name: <String as From<&str>>::from(command_name.into()),
```
Comment (joncinque) [inline token/cli/src/main.rs:3099]: nit: how about:
```suggestion
        command_name: command_name.to_string()
```
Should be possible according to https://docs.rs/strum/latest/strum/derive.ToString.html

**Review by joncinque — COMMENTED** [review]
Just two little nits, then this is good to go!

File: token/cli/src/main.rs (hunk 5 above)
Comment (atharmohammad) [reply token/cli/src/main.rs]: Didn't knew this, cool :)

**Comment by joncinque** [comment]
The changes look good, so now we'll have to get it through CI :smile: 

First, can you reformat the code by running `cargo fmt --all` at the root of the repo?

Second, can you rebase your branch onto master? We had to bump a crate in order to pass the cargo audit as part of CI.

**Comment by atharmohammad** [comment]
> The changes look good, so now we'll have to get it through CI smile
> 
> First, can you reformat the code by running `cargo fmt --all` at the root of the repo?
> 
> Second, can you rebase your branch onto master? We had to bump a crate in order to pass the cargo audit as part of CI.
Done the formatting part, was confused why does do-audit.sh fails thanks for letting me know

**Comment by joncinque** [comment]
Great work! Thanks for the contribution

**Pull request merged**